scipy/sparse/linalg/_isolve/tests/test_lgmres.py

```diff
@@ -100,16 +100,15 @@ def test_outer_v(self):
     def test_arnoldi(self):
         np.random.seed(1234)
 
-        A = eye(2000) + rand(2000, 2000, density=5e-4)
+        A = eye_array(2000) + random_array((2000, 2000), density=5e-4)
+
         b = np.random.rand(2000)
 
         # The inner arnoldi should be equivalent to gmres
         with suppress_warnings() as sup:
             sup.filter(DeprecationWarning, ".*called without specifying.*")
-            x0, flag0 = lgmres(A, b, x0=zeros(A.shape[0]),
```

Comment:
These parameter value changes allow convergence for a variety of seed values. The switch from `rand` to `random_array` effectively changed the seed causing non-convergence for `m=15`.  Switching to `m=10` makes this test more robust to changes in seeds or random number generation. (similar to comment above)